data/src/main/java/org/apache/iceberg/data/GenericReader.java

```diff
@@ -126,6 +128,13 @@ private CloseableIterable<Record> openFile(FileScanTask task, Schema fileProject
           parquet.reuseContainers();
         }
 
+        if (task.file().keyMetadata() != null) {
+          EncryptionKeyMetadata keyMetadata =
```

Comment:
This method in `EncryptionUtil` returns a package-private implementation, `KeyMetadata`. That leaks the class outside of the package but it isn't useful. I think that's why this added methods to `EncryptionKeyMetadata`. I think it's better to make `KeyMetadata` (or `StandardKeyMetadata` as I renamed it in my last PR) public.

Reply:
SGTM